File: app/src/main/java/edu/gatech/cs2340/spacetraderredux/Model/SolarSystem.java
package edu.gatech.cs2340.spacetraderredux.Model;

import android.graphics.Point;

import java.util.ArrayList;
import java.util.Collections;
import java.util.HashSet;
import java.util.Random;
import java.util.Set;

public class SolarSystem {
    private final SolarSystemName name;
    private final Point location;
    private final Planet[] planets;

    public SolarSystem(SolarSystemName name, Point location, Random seed){
        this.name = name;
        this.location = location;
        this.planets = new Planet[seed.nextInt(10) + 1];
        for (int i = 0; i < this.planets.length; i++) {
            this.planets[i] = new Planet(seed);
        }
    }

    @Override
    public String toString() {
        StringBuilder sb = new StringBuilder();
        sb.append("Solar System " + name.name() + " at (" + location.x + ", " + location.y + ") containing planets: {");
        for (int i = 0; i < planets.length - 1; i++) {
            sb.append(planets[i] + ", ");
        }
        sb.append(planets[planets.length - 1] + "}");
        return sb.toString();
    }
}

File: app/src/main/java/edu/gatech/cs2340/spacetraderredux/domain/SolarSystem.kt
package edu.gatech.cs2340.spacetraderredux.domain

import edu.gatech.cs2340.spacetraderredux.domain.entities.enums.SolarSystemName

import java.util.Random

class SolarSystem(val name: SolarSystemName, val location: Point, rand: Random) {
    val planets: Array<Planet>

    init {
        var set = HashSet<String>()
        this.planets = Array(rand.nextInt(10) + 1){generateRandomPlanet(rand)}
        for (i in 0..planets.size-1) {
            var currPlanet = generateRandomPlanet(rand)
            while (set.contains(currPlanet.name.name)) {
                currPlanet = generateRandomPlanet(rand)
            }
            set.add(currPlanet.name.name)
        }
    }

    private fun generateRandomPlanet(rand: Random): Planet {
        return Planet(rand);
    }

    override fun toString(): String {
        val sb = StringBuilder()
        sb.append("Solar System " + name.displayName + " at (" + location.x + ", " + location.y + ") containing planets: {")
        for (i in 0 until planets.size - 1) {
            sb.append(planets[i].toString() + ", ")
        }
        sb.append(planets[planets.size - 1].toString() + "}")
        return sb.toString()
    }
}
